static/app/views/monitors/components/monitorForm.tsx

```diff
@@ -85,8 +85,20 @@ interface TransformedData extends Partial<Omit<Monitor, 'config' | 'alertRule'>>
 /**
  * Transform sub-fields for what the API expects
  */
-function transformData(_data: Record<string, any>, model: FormModel) {
-  const result = model.fields.toJSON().reduce<TransformedData>((data, [k, v]) => {
+export function transformData(_data: Record<string, any>, model: FormModel) {
+  const schedType = model.getValue('config.schedule_type');
+  // Remove interval fields if the monitor schedule is crontab
+  const filteredFields = model.fields
```

Comment:
needed to add this because the new monitor form has both crontab configuration and interval configuration which will be sent to this method. Then I filter out to the corresponding schedule fields (because interval takes priority I only need to remove the interval fields when crontab is selected)